craft_application/commands/lifecycle.py

```diff
@@ -204,7 +200,7 @@ def _run(
         try:
             self._services.lifecycle.run(
                 step_name=step_name,
-                part_names=parsed_args.parts,
+                part_names=parts,
```

Comment:
I think this line is a bit incongruous with the class' docstring of "not needing a part".
Here's my suggestion: how about adding a `def _run_lifecycle(parsed_args)` method that defaults to `self._services.lifecycle.run(step_name)` here but is overridden to add the parts in `LifecyclePartsCommand`?

Reply:
Great idea!